hal_st/stm32fxxx/UartStmDuplexDma.hpp

```diff
@@ -1,42 +1,28 @@
 #ifndef HAL_UART_STM_DUPLEX_DMA_HPP
 #define HAL_UART_STM_DUPLEX_DMA_HPP
 
-#include "hal/interfaces/SerialCommunication.hpp"
-#include "hal_st/cortex/InterruptCortex.hpp"
 #include "hal_st/stm32fxxx/DmaStm.hpp"
-#include "hal_st/stm32fxxx/GpioStm.hpp"
+#include "hal_st/stm32fxxx/UartStm.hpp"
 #include <atomic>
 #include <cstdint>
 
-#include DEVICE_HEADER
-
 namespace hal
 {
-    namespace detail
-    {
-        struct UartStmDuplexDmaConfig
-        {
-            uint32_t baudrate{ 115200 };
-            hal::InterruptPriority priority{ hal::InterruptPriority::Normal };
-
-#if defined(UART_ADVFEATURE_SWAP_INIT)
-            bool swapTxRx{ false };
-#endif
-        };
-    }
-
     class UartStmDuplexDma
-        : public SerialCommunication
-        , private InterruptHandler
+        : public UartStm
     {
     public:
-        using Config = detail::UartStmDuplexDmaConfig;
+        using Config = detail::UartStmConfig;
 
         template<std::size_t RxBufferSize>
         using WithRxBuffer = infra::WithStorage<UartStmDuplexDma, std::array<uint8_t, RxBufferSize>>;
 
         UartStmDuplexDma(infra::MemoryRange<uint8_t> rxBuffer, hal::DmaStm::TransmitStream& transmitStream, hal::DmaStm::ReceiveStream& receiveStream, uint8_t oneBasedIndex, GpioPinStm& uartTx, GpioPinStm& uartRx, const Config& config = Config());
         UartStmDuplexDma(infra::MemoryRange<uint8_t> rxBuffer, hal::DmaStm::TransmitStream& transmitStream, hal::DmaStm::ReceiveStream& receiveStream, uint8_t oneBasedIndex, GpioPinStm& uartTx, GpioPinStm& uartRx, GpioPinStm& uartRts, GpioPinStm& uartCts, const Config& config = Config());
+#if defined(HAS_PERIPHERAL_LPUART)
```

Comment:
Actually this is not supported, as the LPUART doesn't have a timeout interrupt

Reply:
Does that mean the following? :
- Devices on low power mode will not wakeup because of no interrupt
- No wake up = no RX

Reply:
Not really. Wake-up should work regardless DMA on RX.

Timeout interrupt is used to detect when RX is idle.
Without the timeout interrupt, only the DMA interrupts will fire. That is ok for applications where RX is continuously receiving data (no idling at all)

Reply:
Thanks for the information! It was very helpful!
I have considered timer interrupts as a replacement but I don't think that's a good idea...
I concluded I will apply the same principles to UartStmDma and refactor that class. Then UartStmDuplexDma will inherit from that new class and the LPUART constructors will be removed from the duplex version.